### PR TITLE
ontology docker support

### PR DESCRIPTION
**Pull request by Honglei-Cong**
ontology docker is built in two step:
1. build ontology in docker golang:1.9.5-stretch
2. build ontology docker based on ubuntu-core:14.04

golang:1.9.5-stretch is a minimal go-build-env docker
ubuntu-core:14.04 is a minimal usable host docker.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,12 +5,57 @@ Minversion := $(shell date)
...
+DOCKER_TAG=$(ARCH)-$(VERSION)
+ONT_CFG_IN_DOCKER=config-solo.json
+WALLET_FILE=wallet.dat
+
+all: ontology nodectl
+
+ontology:
 	$(GC)  $(BUILD_NODE_PAR) -o ontology main.go
+
+nodectl:
```
Comment (Arbio5zt) [inline Makefile]: nodectl has been removed in version v0.7.0

File: Makefile [hunk 2]
```diff
@@ -5,12 +5,57 @@ Minversion := $(shell date)
...
+DBUILD=docker build
+DRUN=docker run
+DOCKER_NS ?= ontio
+DOCKER_TAG=$(ARCH)-$(VERSION)
+ONT_CFG_IN_DOCKER=config-solo.json
+WALLET_FILE=wallet.dat
+
+all: ontology nodectl
+
+ontology:
```
Comment (Arbio5zt) [inline Makefile]: Is there a step need to glide install.

File: Makefile (hunk 1 above)
Comment (Honglei-Cong) [reply Makefile]: I found 'make all' will still build nodectl.  
I'll update the pull-request when Makefile updated.

File: Makefile (hunk 2 above)
Comment (Honglei-Cong) [reply Makefile]: "glide install" still need to be done by user after 'git clone'
As "make all", "make docker" also just build ontology for docker.

**Review by laizy — APPROVED** [review]

**Pull request merged**